fecgov/openFEC
# Create load_archived_murs and extract_pdf_text commands

## Summary (required)
Currently, archived MUR data sources come from many places: 
1)murArchive.xml, 
2)mur_name.csv (github openFEC repo), 
3)mur_arch.subjct_from_file (oracle FECP)
4)pdf OCR text (on Elasticsearch)
<img width="432" alt="Screen Shot 2020-08-28 at 1 12 16 AM" src="https://user-images.githubusercontent.com/24395751/91523790-a0ae3d00-e8cb-11ea-953e-b714e2f4e1e8.png">






This PR will achieve: (required reviewers: Laura, Priya.  optional reviewers: Jean or anyone)
- [x] centralize all archived MUR data resources into Arouca database. (convert and import data)
- [x] write python code to extract pdf text from Elasticsearch insert to database
- [x] write new python code to upload archived MUR on Elasticsearch (data from database)
- [ ] test case for upload archived MUR, will create separated ticket. 


- Resolves #4490  #4465  #4572

_Include a summary of proposed changes._
add two commands:
1)load_archived_murs
2)extract_pdf_text

## How to test the changes locally
=========(start) Test locally========
1) Install Elasticsearch5.6 locally
a) Check Java version 1.8.0_16 (`java -version`)
b) don't install ES global (I install parallel with openFEC folder)
c) run these two commands to install
`curl -L -O https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.6.8.tar.gz`
`tar -xvf elasticsearch-5.6.8.tar.gz`
d) start Elasticsearch
`cd elasticsearch-5.6.8/bin`
`./elasticsearch `
e) check elasticsearch server: [http://127.0.0.1:9200/](http://127.0.0.1:9200/)

2) check out feature branch
point to dev db

3) create es-index, alias, mapping (run commands on terminal)
a) create index: docs (we don't need for this PR, just practice): 
`./manage.py create_docs_index`
b) create index: archived_murs (this PR will use) 
`./manage.py create_archived_murs_index`

4) check you local Elasticsearch on browser.
a) two indexes : [http://localhost:9200/_cat/indices?v](http://localhost:9200/_cat/indices?v)
b) aliases under each index: [http://localhost:9200/_aliases?pretty](http://localhost:9200/_aliases?pretty)
c) mappings under each index: [http://localhost:9200/_mappings](http://localhost:9200/_mappings)

5) if you want display good JSON format of data: 
    uncomment webservices/legal_docs/archived_murs.py 
line 17: # logger.setLevel(logging.DEBUG)

6) run: upload **single archived mur** on terminal:
`python manage.py load_archived_murs -m 179`
`python manage.py load_archived_murs -m 1503`

7) check archived_mur you loaded on browser, make sure all correct JSON format and text clean.
(single pdf:)
[http://localhost:9200/archived_murs/murs/mur_179/?pretty](http://localhost:9200/archived_murs/murs/mur_179/?pretty)

(multi pdf:)
[http://localhost:9200/archived_murs/murs/mur_1503/?pretty](http://localhost:9200/archived_murs/murs/mur_1503/?pretty)

(total archived_murs count:)
[http://localhost:9200/archived_murs/_count?q=*:*](http://localhost:9200/archived_murs/_count?q=*:*)

7) run: upload **all archived mur** on terminal (take about 3.5 hours, you can ctr+c any time):
`python manage.py load_archived_murs`

(total archived murs: 4448)
=========(end) Test locally========

=========Test on space========
I already upload all arch-murs on stage (headless app), not sure needs test one more time.
[https://api-stage.open.fec.gov/v1/legal/search/?case_no=179&type=murs&api_key=DEMO_KEY](https://api-stage.open.fec.gov/v1/legal/search/?case_no=179&type=murs&api_key=DEMO_KEY)

[https://api-stage.open.fec.gov/v1/legal/search/?case_no=1503&type=murs&api_key=DEMO_KEY](https://api-stage.open.fec.gov/v1/legal/search/?case_no=1503&type=murs&api_key=DEMO_KEY)

===========
The **extract_pdf_text** command is one time job. run it carefully.
you can modify **INSERT_DOCUMENT** to point other table, like DOCUMENTS_X.
